src/utils/bindActionCreators.ts

```diff
@@ -0,0 +1,25 @@
+import { ActionCreator, ActionCreatorsMapObject, AnyAction, Dispatch } from "redux"
```

Comment:
I read somewhere that `bindActionCreators` was previously copied from redux itself so I've done the same thing to make it TS-compatible: https://github.com/reduxjs/redux/blob/master/src/bindActionCreators.ts

Reply:
Yep, we inlined this into the React-Redux codebase in one of the last couple patch releases.

Reply:
I think we can split the difference and just do a TS-ified version of the plain JS file we had in here. I don't think we need the separate single-function version for our use case here.

Reply:
That was my initial approach but it causes a compilation error on `const actionCreator = actionCreators[key]`. 

By splitting the code path using `if (typeof actionCreators === 'function')`, `actionCreators` is understood to only be `ActionCreatorsMapObject<any>` from that point onwards. Without the split, it is `ActionCreator<any> | ActionCreatorsMapObject<any>` which does not support the indexer. 

I could inline the `bindActionCreator()` function but figured it would be easier for future reference if the code is more of a straight-up copy. Happy to change it inline (or to some other solution), whichever you prefer.

Reply:
@markerikson Not sure if you missed the above message ^ Do you have any thoughts on the path to take here?

Reply:
Yep, missed it :)

Let's just go with the copy-pasted version for now, then, and we can always re-tweak it later.